Sources/StreamVideo/WebRTC/v2/PeerConnection/Adapters/ICEAdapter.swift

```diff
@@ -222,25 +220,22 @@ actor ICEAdapter: @unchecked Sendable {
     ///   regardless of success or failure of individual additions.
     private func drainPendingSFUCandidates() async {
         let candidates = pendingSFUCandidates
-        await withTaskGroup(of: Void.self) { [weak self] group in
-            guard let self else { return }
-            for candidate in candidates {
-                group.addTask { [weak self] in
-                    guard let self else { return }
-                    do {
+        for candidate in candidates {
+            Task(disposableBag: disposableBag) { [weak self] in
```

Comment:
are we ok with fire and forget behaviour here?

Reply:
It's not forget though. The tasks are stored inside the disposable bag. We have control over 1. when to cancel them if they are still inflight, 2. dispose them once they complete.